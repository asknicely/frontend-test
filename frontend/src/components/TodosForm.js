import React from 'react'

const Form = (props) => {
    return (
        <tr>
            <td colSpan="4" align="left">
                <input 
                    type="textbox" 
                    name="description" 
                    className="small-6 small-center" 
                    placeholder="Description..." 
                    value={props.description}
                    onChange={(e) => props.setDescription(e.target.value) }
                />
            </td>
            <td>
                <button 
                    type="submit" 
                    className="btn btn-sm btn-primary"
                    onClick={props.handleSubmit}
                >
                Add
                </button>
            </td>
        </tr>
    )
}

export default Form