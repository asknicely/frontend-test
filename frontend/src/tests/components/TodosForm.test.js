import React, { useReducer } from 'react';
import { shallow } from 'enzyme';
import TodosForm from '../../components/TodosForm';

test('should render TodosForm correctly', () => {

	const item = 2
	const wrapper = shallow(
		<TodosForm 
			description=""
			setDescription={(e)=>console.log(e.target.value)}
			handleSubmit={(e)=>console.log('handleSubmit')}		
		/>
	);

	expect(wrapper).toMatchSnapshot();
	
});
