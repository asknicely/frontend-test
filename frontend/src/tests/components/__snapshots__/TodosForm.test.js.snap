// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TodosForm correctly 1`] = `
<tr>
  <td
    align="left"
    colSpan="4"
  >
    <input
      className="small-6 small-center"
      name="description"
      onChange={[Function]}
      placeholder="Description..."
      type="textbox"
      value=""
    />
  </td>
  <td>
    <button
      className="btn btn-sm btn-primary"
      onClick={[Function]}
      type="submit"
    >
      Add
    </button>
  </td>
</tr>
`;
